Finished with HiCdat tool
=========================

[Reference: HiCdat](http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0678-x)  [github](https://github.com/MWSchmid/HiCdat)

###Pre-install

[R](https://www.r-project.org/)</br>
[Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)</br>
[Samtools](http://samtools.sourceforge.net/)</br>
[HiCdatPre Binary for MacOSX 64bit](https://github.com/MWSchmid/HiCdat/blob/master/other/mac_64bit.zip?raw=true)</br>
[HiCdatPre Binary for windows 7 64bit](https://github.com/MWSchmid/HiCdat/blob/master/other/windows_64bit.zip?raw=true)</br>
[HiCdatPre Binary for linux 64bit (need Qt4 lib)](https://github.com/MWSchmid/HiCdat/blob/master/other/linux_64bit.zip?raw=true)</br>

HiCdatR requires the R libraries "gplots", "randomizeBE", "MASS", and "HiCseg". You can install them with:
```r
install.packages(c("gplots", "randomizeBE", "MASS"))
source("http://bioconductor.org/biocLite.R")
biocLite("HiCseg")

install.packages("/path/to/HiCdatR_0.99.0.tar.gz", repos=NULL, type = "source")
```


###Download Project
```
git clone https://github.com/yad50968/NCCU_Bioinformatics_Final.git
cd NCCU_Bioinformatics_Final/HiCdat/
```

###Step1 
Download fastq and Decompress (It will create fastq folder)<br>
About need 150GB
```sh
sh ./download_fastq.sh
```


###Step2 
Download fasta, Decompress, filter it (It will create fasta folder)
```sh
sh ./download_fasta.sh
```

###Step3 
Use Bowtie2 to build index
```sh
bowtie2-build ./fasta/fasta_filtered.fasta ./fasta_index/build_fasta
```

###Step4 
Use Bowtie2 and Samtool to generate .BAM
```sh
sh run_bowtie_samtool.sh
```

###Step5 
Use HiCdatPre to merge BAM files<br><br>
###Step6 
Use HiCdatPre to generate fragments (Set binsize : 100000) ex: ./fasta_fra.txt</br></br>
###Step7 
Use HiCdatPre to generate map read-pairs to fragments ex ./Data/<br></br>
###Step8 
Use HiCdatPre to generate R-source file containing basic genome information of organism<br>ex ./Rscripts/organism-specific_R-code.R)</br></br>

###Step9 
Fix the path and the file names in the Rscript/HiCdat-Result.R <br>
###Step10 
Run
```sh
Rscript ./Rscript/HiCdat-Result.R
```

